assignment6/rearranging_cars.py

```diff
@@ -0,0 +1,70 @@
+'''
+Created on 07 July 2017
+
+@author: Eva
+'''
+
+def print_car_moves(initial_car_park, final_car_park):
+    car_moves = rearrange_cars(initial_car_park, final_car_park)
+    
+    for i in range(len(car_moves)):
+        print(car_moves[i].__str__())
+
+def rearrange_cars(initial_car_park, final_car_park):
+    """Rearranges the cars from the initial positions to the end positions.
+    
+    Args:
+        initial_car_park: an array, represents the initial positions of the cars in the parking lot
+        final_car_park: an array, represents the final positions of the cars in the parking lot
+        
+    Returns:
+        an array of Move objects that represent the moves taken to rearrange the cars to their final positions
+    """
+    if len(initial_car_park) != len(final_car_park):
+        raise ValueError("must be the same length: initial_car_park, final_car_park")
+    
+    EMPTY_SPACE = 0
+    # Dictionary used for O(1) lookups when searching for the index of a value in the car_park
+    car_map = {}
```

Comment:
By the way, you can use an ordinary list for this, because all elements has indexes `0`..`N-1`. From my opinion, it's better, as long as hash-table takes more space than list.